resources/nspawn.go

```diff
@@ -0,0 +1,351 @@
+// Watch for state changes and sends a message to the bus if there is a change.
+func (obj *NspawnRes) Watch(processChan chan event.Event) error {
+	// Boilerplate
+	// TODO: Boilerplate sucks. Find a way to prevent its necessity.
+	if obj.IsWatching() {
+		return nil
+	}
+	obj.SetWatching(true)
+	defer obj.SetWatching(false)
+	cuid := obj.converger.Register()
+	defer cuid.Unregister()
+
+	var startup bool
+	Startup := func(block bool) <-chan time.Time {
+		if block {
+			return nil // blocks forever
+		}
+		return time.After(time.Duration(500) * time.Millisecond) // 1/2 the resolution of converged timeout
+	}
+	// End Boilerplate
+
+	// This resource depends on systemd. Ensure that it's running.
+	// TODO: If systemd is not running, should we start it?
+	if !systemdUtil.IsRunningSystemd() {
+		return fmt.Errorf("Systemd is not running.")
+	}
+
+	// Create a private message bus
+	bus, err := util.SystemBusPrivateUsable()
+	if err != nil {
+		return fmt.Errorf("Failed to connect to bus: %s", err)
+	}
+
+	// Add a match rule to match messages going through the message bus.
+	bus.BusObject().Call("org.freedesktop.DBus.AddMatch", 0,
+		"type='signal',interface='org.freedesktop.machine1.Manager',eavesdrop='true'")
+	buschan := make(chan *dbus.Signal, 10)
+	bus.Signal(buschan)
+
+	var send = false
+	var exit = false
+	// var dirty = false
+	// var invalid = false
+	// var activeSet = false
+	// var previous bool
+
+	for {
+		obj.SetState(ResStateWatching)
+		select {
+		// If this resource has been idle for long enough, set converged to allow
+		// the main loop to act on a converged status.
+		case <-cuid.ConvergedTimer():
+			cuid.SetConverged(true) // converged!
+			continue
+
+		case event := <-obj.Events():
+			cuid.SetConverged(false)
+			// we avoid sending events on unpause
+			if exit, send = obj.ReadEvent(&event); exit {
+				return nil // exit
+			}
+
+		case <-Startup(startup):
+			cuid.SetConverged(false)
+			send = true
+
+		// Process org.freedesktop.machine1 events for this resource's name.
+		case event := <-buschan:
+			if event.Body[0] == obj.GetName() {
+				if event.Name == machineNew {
+					log.Printf("Machine Started: %#v", event.Body[0])
+				}
+				if event.Name == machineRemoved {
+					log.Printf("Machine Stopped: %#v", event.Body[0])
+				}
+			}
+			send = true
+		}
+
+		// do all our event sending all together to avoid duplicate msgs
+		if send {
+			startup = true // startup finished
+			send = false
+			// only do this on certain types of events
+			//obj.isStateOK = false // something made state dirty
+			if exit, err := obj.DoSend(processChan, ""); exit || err != nil {
+				return err // we exit or bubble up a NACK...
+			}
+		}
+	}
+}
+
+// CheckApply is run to check the state and, if apply is true, to apply the necessary changes to reach the desired
+// state. This is run before Watch and again if watch finds a change occurring to the state.
+func (obj *NspawnRes) CheckApply(apply bool) (checkok bool, err error) {
```

Comment:
We don't seem to look at the apply variable anywhere... It's required, see other resources for how it's used.